### ISSUE TITLE
ERROR notification handler failed handler=textDocument/didSave error=file not found

### ISSUE THREAD

**Issue by lnicola** [rust-lang/rust-analyzer#18747]
`~/Projects/rust-analyzer/Cargo.toml`

Spotted in 27e824fad4 2024-12-15

**Comment by Aisuko** [comment]
I hit same issue on Ubuntu 24.04

```2025-01-30T08:18:15.333161679Z ERROR notification handler failed handler=textDocument/didSave error=file not found: /workspaces/speed/Cargo.toml```

**Comment by spector32** [comment]
Also an issue in WSL Ubuntu 22.04.5 LTS. Running the `chmod 0777 . -R` on all the project files did help, however, I doubt that it can be called a solution 😆

**Comment by lnicola** [comment]
It wasn't a permission issue in my case, but it was gone after restarting the editor.

**Comment by lflys** [comment]
I also encountered the same issue on Windows 11:

```
2025-12-01T10:46:06.0093393+08:00 ERROR notification handler failed handler=textDocument/didSave error=file not found: d:\DevProj\xxx\Cargo.toml
```

This issue occurs when saving `Cargo.toml` in the project root directory, but when updating `Cargo.toml` in a member directory, Cargo checks out normally. Has anyone else encountered a similar issue on Windows?

I looked at issue #19528, but that's not the cause of the problem I'm facing.

**Comment by scmccarthy** [comment]
> This issue occurs when saving `Cargo.toml` in the project root directory, but when updating `Cargo.toml` in a member directory, Cargo checks out normally. Has anyone else encountered a similar issue on Windows?

Yeah I just noticed the same thing. Windows 11. Vscode with rust-analyzer extension.

My root Cargo.toml looks like this:
```
[workspace]
members = ["www", "xxx", "yyy", "zzz"]
resolver = "3"

[workspace.package]
version = "0.1.0"
edition = "2024"

[workspace.dependencies]
thiserror = "2"
rand = "0.9"
```

Haven't noticed anything else working incorrectly.

**Comment by jhead** [comment]
I encountered the same issue with the Cargo.toml in a subdir from the project root. Directly opening that subdir in vscode/cursor worked fine.

```
project/
├── my_crate/
│   ├── Cargo.toml
│   └── ...
```

I found a workaround: using **File** -> **Add Folder to Workspace** to add the subdir (e.g. `my_crate`) to the workspace resolves the issue for me. You end up with the folder in there twice but oh well.
